include/drivers/flash.h

```diff
@@ -293,7 +304,30 @@ static inline size_t z_impl_flash_get_write_block_size(struct device *dev)
 	const struct flash_driver_api *api =
 		(const struct flash_driver_api *)dev->driver_api;
 
-	return api->write_block_size;
+	return api->get_parameters(dev)->write_block_size;
+}
+
+
+/**
+ *  @brief  Get pointer to flash_parameters structure
+ *
+ *  Returned pointer points to a structure that should be considered
+ *  constant through a runtime, regardless if it is defined in RAM or
+ *  Flash.
+ *  Developer is free to cache the structure pointer or copy its contents,
+ *  although a pointer if preferred method of access.
```

Comment:
I'd make this end at "copy its contents."  The rest is opinion that isn't good advice for some uses.  If you want it anyway just change "if preferred" to "is the preferred".

Reply:
Done